packages/ListBox/src/components/Filter/Filter.js

```diff
@@ -0,0 +1,131 @@
+  value: null,
+};
+
+export default function Filter(props) {
+  const [state, dispatch] = useListBox();
+  const [textSearch, setTextSearch] = React.useState(props.value);
+  const applyFilterType = useListBox.types.applyFilter;
+
+  const handleChangeFilter = event => {
+    function hasNoResults(textSearchValue, filteredOptions) {
+      return textSearchValue && (filteredOptions && filteredOptions.length === 0);
+    }
+
+    const textSearchValue = event.target.value;
+
+    if (state.isDisabled) return;
+
+    if (props.filter) {
+      setTextSearch(textSearchValue);
+
+      const result = props.filter({ search: textSearchValue });
+      if (result && result.then) {
+        // detecting if it's a promise
+        result.then(response => {
+          const filteredOptions = response;
+          const noResultsFound = hasNoResults(textSearchValue, filteredOptions);
+          applyFilter(dispatch, applyFilterType)(filteredOptions, noResultsFound);
+        });
+      }
+      return;
+    }
+
+    if (props.onChangeFilter) {
+      props.onChangeFilter(event);
+    } else {
+      setTextSearch(textSearchValue);
+    }
+
+    const filteredOptions = filter({ props, state, textSearchValue });
+    const noResultsFound = hasNoResults(textSearchValue, filteredOptions);
+    applyFilter(dispatch, applyFilterType)(filteredOptions, noResultsFound);
+  };
+
+  const handleBlur = () => {
+    window.requestAnimationFrame(() => {
+      if (state.hasFooter) {
+        // if has footer the responsible of closing is the Footer no the onblur event
+        return;
+      }
+
+      if (document.activeElement !== state.refListBoxContainer.current) {
+        // this will reset the activeOption and close the Popover
+        dispatch({
+          type: useListBox.types.setActiveOption,
+          payload: {
+            activeOptionIndex: null,
+          },
+        });
+      }
+    });
+  };
+
+  const handleEffectValue = effects.handleEffectValue(props, applyFilter(dispatch, applyFilterType));
+  const handleEffectIsPopOverOpen = effects.handleEffectIsPopOverOpen(state, setTextSearch);
+  const handleEffectTextSearch = effects.handleEffectTextSearch(textSearch, applyFilter(dispatch, applyFilterType));
+  const handleEffectHasFilter = effects.handleEffectHasFilter(dispatch, useListBox.types.hasFilter);
+
+  React.useEffect(handleEffectValue, [props.value]);
+  React.useEffect(handleEffectIsPopOverOpen, [state.isOpen]);
+  React.useEffect(handleEffectTextSearch, [textSearch]);
+  React.useEffect(handleEffectHasFilter, []);
+
+  if (state.isInline || state.isOpen) {
+    const { renderFilter, placeholder, value, onChangeFilter, filter, ...moreProps } = props;
+    if (renderFilter) {
+      return props.renderFilter(props);
+    }
+
+    return (
+      <FilterContainerStyled data-qa-anchor="list-filter">
+        {props.hasSearchIcon ? <FilterSearchIconStyled /> : null}
+        <FilterInputStyled
+          data-qa-anchor="list-filter-input"
```

Comment:
A BEM-style `listbox-filter__input` might be a better name?

Reply:
is not css IMO and haven't seen using BEM for this in any project at ecosystem level so far. But if we want to go with the BEM for paprika, I'm not really opposed. 

But I think this is more a standard decision at repo level than in this PR